src/main/java/com/streisky/discordchatgptbot/utils/JsonUtils.java

```diff
@@ -0,0 +1,10 @@
+package com.streisky.discordchatgptbot.utils;
+
+import com.google.gson.Gson;
+
+public class JsonUtils {
+
+    public static <T> JsonCompatible convertJsonToObject(Class<? extends JsonCompatible> c, String json) {
+        return new Gson().fromJson(json, c);
```

Comment:
Instead of creating the Gson object here, it might be a good idea to inject it through dependency injection.

Reply:
Not worth it for this project.